src/dialogs/more-info/controls/more-info-weather.ts

```diff
@@ -13,26 +13,119 @@ import {
   PropertyValues,
   nothing,
 } from "lit";
-import { customElement, property } from "lit/decorators";
+import { customElement, property, state } from "lit/decorators";
+import { HassEntityBase } from "home-assistant-js-websocket";
 import { formatDateWeekdayDay } from "../../../common/datetime/format_date";
 import { formatTimeWeekday } from "../../../common/datetime/format_time";
 import { formatNumber } from "../../../common/number/format_number";
 import "../../../components/ha-svg-icon";
 import {
+  getForecast,
   getWeatherUnit,
   getWind,
-  isForecastHourly,
+  subscribeForecast,
+  ForecastEvent,
   WeatherEntity,
+  WeatherEntityFeature,
   weatherIcons,
 } from "../../../data/weather";
 import { HomeAssistant } from "../../../types";
+import { supportsFeature } from "../../../common/entity/supports-feature";
 
 @customElement("more-info-weather")
 class MoreInfoWeather extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
   @property() public stateObj?: WeatherEntity;
 
+  @state() private _forecastEvent?: ForecastEvent;
+
+  @state() private _subscribed?: Promise<() => void>;
+
+  private _needForecastSubscription() {
+    return supportsFeature(
+      this.stateObj as HassEntityBase,
+      // eslint-disable-next-line no-bitwise
+      WeatherEntityFeature.FORECAST_DAILY |
+        // eslint-disable-next-line no-bitwise
+        WeatherEntityFeature.FORECAST_HOURLY |
+        WeatherEntityFeature.FORECAST_TWICE_DAILY
+    );
+  }
+
+  private _forecastType() {
+    if (
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_DAILY
+      )
+    ) {
+      return "daily";
+    }
+    if (
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_HOURLY
+      )
+    ) {
+      return "hourly";
+    }
+    if (
+      supportsFeature(
+        this.stateObj as HassEntityBase,
+        WeatherEntityFeature.FORECAST_TWICE_DAILY
+      )
+    ) {
+      return "twice_daily";
+    }
+    return undefined;
+  }
+
+  private _handleForecastEvent(forecastEvent: ForecastEvent) {
+    this._forecastEvent = forecastEvent;
+  }
+
+  private _unsubscribeForecastEvents() {
+    if (this._subscribed) {
+      this._subscribed.then((unsub) => unsub());
+      this._subscribed = undefined;
+    }
+  }
+
+  private async _subscribeForecastEvents() {
+    if (
+      this._subscribed ||
+      !this.stateObj ||
+      !this.hass ||
+      !this.isConnected ||
+      !this._needForecastSubscription()
+    ) {
+      return;
+    }
+
+    const forecastType = this._forecastType();
+    if (forecastType) {
+      this._subscribed = subscribeForecast(
+        this.hass!,
+        this.stateObj!.entity_id,
+        forecastType,
+        (event) => this._handleForecastEvent(event)
```

Comment:
You can inline this function as it's just 1 line. 
```suggestion
        (event) => { this._forecastEvent = event; }
```

Reply:
Makes sense, fixed